### PR TITLE
Sdx collect

### PR DESCRIPTION
**Pull request by dgunthor**
refactor perkin to be sdx-collect - retrieve survey from q, parse, store, receipt

test - anyone but david gunthorpe once sdx-store is ready

### CODE REVIEW

**Comment by howellsaj** [comment]
Chatting through the design with @iwootten and how we thought that the sdx-collect service should now only be interested in getting json, validating and passing it on to sdx-store. This code currently, still parsers the json into a Survey object which it doesn't need to do as this is now probably a function that the sdx-transform-cs service carries out. Any thoughts?

**Comment by dgunthor** [comment]
sounds good, i'll update the changeset...

File: src/main/java/com/github/onsdigital/perkin/helper/SdxStore.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        host = ConfigurationManager.get("STORE_HOST");
+        path = ConfigurationManager.get("STORE_PATH");
+    }
+
+    public Response<Result> store(final String json) throws IOException {
+
+        Endpoint endpoint = new Endpoint(new Host(host), path);
+        log.info("STORE|storing survey to endpoint: {}", endpoint);
+
+        return new Http().postJson(endpoint, json, Result.class);
```
Comment (iwootten) [inline src/main/java/com/github/onsdigital/perkin/helper/SdxStore.java]: The postJson method within Http() in Httpino has a bug in it which returns a quoted string, meaning that the strings are all rejected by validate. I've investigated and postString doesn't have this problem (as we're using it on decrypt). You can see we're we've had to make a similar change in the past here: https://github.com/ONSdigital/sdx-collect/commit/51ad12fbd237cff9c976bf04ded40bcde10ded20

Additionally in order to mock out tests, there's a HttpManager class I was using on the decrypt method. If you can use this that means we can can easily create tests for these endpoints, without needing a bunch of dependencies set up.

File: src/main/java/com/github/onsdigital/perkin/helper/SdxValidate.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        host = ConfigurationManager.get("VALIDATE_HOST");
+        path = ConfigurationManager.get("VALIDATE_PATH");
+    }
+
+    public Response<Result> validate(final String json) throws IOException {
+
+        Endpoint endpoint = new Endpoint(new Host(host), path);
+        log.info("VALIDATE|validating survey to endpoint: {}", endpoint);
+
+        return new Http().postJson(endpoint, json, Result.class);
```
Comment (iwootten) [inline src/main/java/com/github/onsdigital/perkin/helper/SdxValidate.java]: Same issues as above here.

File: src/main/java/com/github/onsdigital/perkin/transform/TransformEngine.java [hunk 3]
```diff
@@ -24,77 +20,42 @@
...
         try {
             decrypt = new Decrypt(data);
             String json = decrypt.getDecrypted();
-            Survey survey = parser.parse(json);
 
-            TransformContext context = createTransformContext(survey);
+            //TODO will this throw an Exception or use a return value?
+            //{"valid": True} 200
+            //{"valid": False} 400?
+            Response<Result> response = validate.validate(json);
```
Comment (iwootten) [inline src/main/java/com/github/onsdigital/perkin/transform/TransformEngine.java]: Can see the todos about response codes here - invalid json will respond with 400, so if that can be correctly handled that would be great.

Comment (dgunthor) [inline src/main/java/com/github/onsdigital/perkin/transform/TransformEngine.java]: No probs, what other codes are possible? 500?

Comment (iwootten) [inline src/main/java/com/github/onsdigital/perkin/transform/TransformEngine.java]: Yup, 200 ok, 400 invalid json, 500 unknown error

Comment (dgunthor) [inline src/main/java/com/github/onsdigital/perkin/transform/TransformEngine.java]: ta

**Comment by iwootten** [comment]
👍  Looks good from here, tested against valid, invalid from console.

**Pull request merged**